dot/network/sync.go

```diff
@@ -192,10 +197,7 @@ func (q *syncQueue) syncAtHead() {
 		prev = curr
 		start := uint64(curr.Number.Int64()) + 1
 		logger.Debug("haven't received new blocks since last check, pushing request", "start", start)
-		q.requestData.Store(start, requestData{
-			sent:     true,
-			received: false,
-		})
+		q.requestData.Delete(start)
```

Comment:
I'm not sure I understand what's happening here.  It seems we're deleting start from the requestData map, but the pushing it back on the map?  It seems to be working, but I don't fully understand.

Reply:
the requestData tracks requests that were sent and received, in the case we are at the head it seems like often we will send out a request and receive an empty response if the node doesn't have info for that block yet. the request processor ignores requests that have already gotten a response, so we need to delete the info for it that says we got a request first. I noticed the node would go into a loop of trying to send a request but thinking it already got the response and thus never syncing